src/csharp/Microsoft.Spark/Sql/SparkSession.cs

```diff
@@ -61,10 +61,40 @@ internal SparkSession(JvmObjectReference jvmObject)
         /// <returns>Builder object</returns>
         public static Builder Builder() => new Builder();
 
-        /// Note that *ActiveSession() APIs are not exposed because these APIs work with a
-        /// thread-local variable, which stores the session variable. Since the Netty server
-        /// that handles the requests is multi-threaded, any thread can invoke these APIs,
-        /// resulting in unexpected behaviors if different threads are used.
+        /// <summary>
+        /// Changes the SparkSession that will be returned in this thread when 
+        /// <see cref="Builder.GetOrCreate"/> is called. This can be used to ensure that a given
+        /// thread receives a SparkSession with an isolated session, instead of the global
+        /// (first created) context.
+        /// </summary>
+        /// <param name="session">SparkSession object</param>
+        public static void SetActiveSession(SparkSession session) =>
```

Comment:
Can we add the version attribute `[Since(Versions.V3_0_0)]` for this API?

Reply:
Nevermind this comment, got confused by #633 and [Pyspark](https://github.com/apache/spark/blob/v3.0.1/python/pyspark/sql/session.py#L266) but I see this has been exposed by Spark since 2.0.0 so please ignore this comment.